### ISSUE TITLE
Why are you finding logits over the turn token vs any other token?

### ISSUE THREAD

**Issue by progamer186** [sgrvinod/chess-transformers#19]
Hey, I'm a bit new to transformers, so I was just wondering why you decided to find logits over the "turn" token in your [CT-E-20](https://github.com/sgrvinod/chess-transformers/blob/972aac30ef72daa8505de2a7ad709995c1b07ee0/chess_transformers/transformers/models.py#L223) model instead of finding logits over any other token such as white_kingside_castling_rights? Does it matter in this case, and what significance does the turn token hold here?

**Comment by sgrvinod** [comment]
Hi, it doesn't _strictly_ matter which token you choose: the model will learn to channel its learnings into the token you choose, whatever it might be. 

But some tokens may be better choices than others. For example, choosing the token for kingside castling rights for white might imply that that token is somehow more special than the other castling tokens, and create an asymettry that you probably do not want in your model. You want your token to be one-of-a-kind, if possible.

Of course, it was always an option to create a special classification token and add it at the beginning of every sequence, but I figured it wouldn't be necessary.

**Comment by sgrvinod** [comment]
Also, among all our candidate tokens, it's probably the most logical choice from a human perspective: to relate the turn token (i.e. whose turn it is) to the move to make in that turn.

**Comment by EaswarGn** [comment]
> Of course, it was always an option to create a special classification token and add it at the beginning of every sequence, but I figured it wouldn't be necessary.
Would it be better to do this (create a special classification token)? If so, how would you do it?

**Comment by EaswarGn** [comment]
> Of course, it was always an option to create a special classification token and add it at the beginning of every sequence, but I figured it wouldn't be necessary.

Would it be better to do this (create a special classification token)? If so, how would you do it?

**Comment by sgrvinod** [comment]
I don't think it would be better than just using the turn token. But I can't say for sure. 

Yes, it would involve prepending a CLS token at the start. This can be done by adding a token embedding for it [here](https://github.com/sgrvinod/chess-transformers/blob/2cec8bb384a4b7d19a3f7483c3956efba5ab77d3/chess_transformers/transformers/modules.py#L395), expanding the positional embeddings from 69 positions to 70 [here](https://github.com/sgrvinod/chess-transformers/blob/2cec8bb384a4b7d19a3f7483c3956efba5ab77d3/chess_transformers/transformers/modules.py#L413), including the token embedding at position 0 [here](https://github.com/sgrvinod/chess-transformers/blob/2cec8bb384a4b7d19a3f7483c3956efba5ab77d3/chess_transformers/transformers/modules.py#L494), and making sure you access the correct indices in the respective models, like [here](https://github.com/sgrvinod/chess-transformers/blob/2cec8bb384a4b7d19a3f7483c3956efba5ab77d3/chess_transformers/transformers/models.py#L388) and [here](https://github.com/sgrvinod/chess-transformers/blob/2cec8bb384a4b7d19a3f7483c3956efba5ab77d3/chess_transformers/transformers/models.py#L563). (I just noticed that the comment in the last link is wrong, it should be "Find logits for _From_ and _To_ squares at each square".)

**Comment by sgrvinod** [comment]
@progamer186 Please let me know if you have any more questions, or if I should close this issue as resolved. Thanks!
